src/linkedlist.ts

```diff
@@ -14,6 +24,60 @@ export function fromArray<A>(values: A[]): Cons<A> {
   return list;
 }
 
+/**
+ * A doubly linked list. Updates are done by mutating. Prepend, append
+ * and remove all run in O(1) time.
+ */
+export class DoubleLinkedList<A> {
+  head: Node<A> | undefined;
+  tail: Node<A> | undefined;
+  append(node: Node<A>): DoubleLinkedList<A> {
+    if (this.head === undefined) {
+      this.head = node;
+    }
+    node.prev = this.tail;
+    node.next = undefined;
+    if (this.tail !== undefined) {
+      this.tail.next = node;
+    }
+    this.tail = node;
+    return this;
+  }
+  remove(node: Node<A>): DoubleLinkedList<A> {
+    if (node.next !== undefined) {
+      node.next.prev = node.prev;
+    }
+    if (node.prev !== undefined) {
+      node.prev.next = node.next;
+    }
+    if (this.head === node) {
+      this.head = node.next;
+    }
+    if (this.tail === node) {
+      this.tail = node.prev;
+    }
+    return this;
+  }
+  *[Symbol.iterator](): IterableIterator<A> {
+    let {head} = this;
+    while (head !== undefined) {
+      yield head.value;
+      head = head.next;
+    }
+  }
+}
+export class Node<A> {
+  public prev: Node<A> | undefined;
+  public next: Node<A> | undefined;
+  constructor(
+    public value: A
```

Comment:
Are the values of `Node`s ever mutated? If not we could make this property `readonly`?

Reply:
it does get mutated when appending or deleting nodes in the list